Mathlib/Order/OrderIsoNat.lean

```diff
@@ -112,24 +113,26 @@ variable (s : Set ℕ) [Infinite s]
 /-- An order embedding from `ℕ` to itself with a specified range -/
 def orderEmbeddingOfSet [DecidablePred (· ∈ s)] : ℕ ↪o ℕ :=
   (RelEmbedding.orderEmbeddingOfLTEmbedding
-        (RelEmbedding.natLt (Nat.Subtype.ofNat s) fun n => Nat.Subtype.lt_succ_self _)).trans
+        (RelEmbedding.natLt (Nat.Subtype.ofNat s) fun _ => Nat.Subtype.lt_succ_self _)).trans
     (OrderEmbedding.subtype s)
 #align nat.order_embedding_of_set Nat.orderEmbeddingOfSet
 
-/-- `nat.subtype.of_nat` as an order isomorphism between `ℕ` and an infinite subset. See also
-`nat.nth` for a version where the subset may be finite. -/
+/-- `Nat.Subtype.ofNat` as an order isomorphism between `ℕ` and an infinite subset. See also
+`Nat.Nth` for a version where the subset may be finite. -/
 noncomputable def Subtype.orderIsoOfNat : ℕ ≃o s := by
-  classical exact
+  classical
+  exact
       RelIso.ofSurjective
         (RelEmbedding.orderEmbeddingOfLTEmbedding
           (RelEmbedding.natLt (Nat.Subtype.ofNat s) fun n => Nat.Subtype.lt_succ_self _))
         Nat.Subtype.ofNat_surjective
 #align nat.subtype.order_iso_of_nat Nat.Subtype.orderIsoOfNat
 
-variable {s}
+--porting note: Added the decidability requirement, I'm not sure how it worked in lean3 without it
```

Comment:
Do you need it for the statements, or for the proofs?

Reply:
For the statements. In particular, `orderEmbeddingOfSet` requires it

Reply:
This is the `classical` leakage in Lean3 again